### PR TITLE
Service offloading migration

### PR DESCRIPTION
**Pull request by Sunwoo-Nam**
This PR includes the below contents.
- Media Capture in Android.
- Optimization for webrtc

This PR excludes the commits for mouse and keyboard events.

### CODE REVIEW

File: chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java [hunk 1]
```diff
@@ -1263,6 +1267,16 @@ public void initializeState() {
         } finally {
             TraceEvent.end("ChromeTabbedActivity.initializeState");
         }
+
+        // This chrome activity will be used only for WebRTCGameStreaming.
```
Comment (Chikdolman) [inline chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java:1271]: Only for Service offloading? I think this permission can be used for general cases.

File: content/renderer/media/webrtc/media_stream_video_webrtc_sink.cc [hunk 2]
```diff
@@ -249,6 +249,11 @@ MediaStreamVideoWebRtcSink::MediaStreamVideoWebRtcSink(
       refresh_interval =
           base::TimeDelta::FromMicroseconds(kLowerBoundRefreshIntervalMicros);
     }
+
+#if defined(SERVICE_OFFLOADING)
+    // For WebRTCGameStreaming, keep fps above 16 unless there are new images.
```
Comment (Chikdolman) [inline content/renderer/media/webrtc/media_stream_video_webrtc_sink.cc:254]: For WebRTCGameStreaming -> For Service Offloading

File: build/android/gyp/lint.py [hunk 3]
```diff
@@ -257,6 +257,8 @@ def _FindInDirectories(directories, filename_filter):
 
 
 def main():
+  # Skip for WebRTCGameStreaming
```
Comment (Chikdolman) [inline build/android/gyp/lint.py:260]: WebRTCGameStreaming -> Service offloading

File: content/browser/android/content_startup_flags.cc [hunk 4]
```diff
@@ -78,6 +82,12 @@ void SetContentCommandLineFlags(bool single_process) {
   // Disable anti-aliasing.
   parsed_command_line->AppendSwitch(
       cc::switches::kDisableCompositedAntialiasing);
+
+#if defined(SERVICE_OFFLOADING)
+  // Prevents the renderer process from being killed for WebRTCGameStreaming.
```
Comment (Chikdolman) [inline content/browser/android/content_startup_flags.cc:87]: WebRTCGameStreaming -> Service offloading

File: content/browser/child_process_launcher_helper_android.cc [hunk 5]
```diff
@@ -60,8 +60,11 @@ void ChildProcessLauncherHelper::BeforeLaunchOnClientThread() {
       << "Unsupported process type: " << process_type;
 
   // Non-sandboxed utility or renderer process are currently not supported.
+  // Don't check Sandbox for WebRTCGameStreaming.
```
Comment (Chikdolman) [inline content/browser/child_process_launcher_helper_android.cc:63]: WebRTCGameStreaming -> Service offloading

**Review by Chikdolman — COMMENTED** [review]
Except for comments itself, LGTM

File: chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java (hunk 1 above)
Comment (Sunwoo-Nam) [reply chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java:1271]: I'm going to add the condition using the variable in the manifest.

File: content/renderer/media/webrtc/media_stream_video_webrtc_sink.cc (hunk 2 above)
Comment (Sunwoo-Nam) [reply content/renderer/media/webrtc/media_stream_video_webrtc_sink.cc:254]: Can I rename this by making an additional patch?
Files you haven't mentioned seem to be modified.

**Review by Chikdolman — APPROVED** [review]
Fix ups from the review will be available in the next patch. LGTM

**Pull request merged**